server/src/main/java/org/opensearch/index/fielddata/FieldDataStats.java

```diff
@@ -106,25 +151,58 @@ public long getEvictions() {
         return this.evictions;
     }
 
+    public long getItemCount() {
+        return this.itemCount;
+    }
+
+    /**
+     * Get the field memory sizes.
+     * Cannot change the method name as this class was marked PublicApi.
+     * @return The memory field sizes
+     */
     @Nullable
     public FieldMemoryStats getFields() {
-        return fields;
+        return fieldMemorySizes;
+    }
+
+    @Nullable
+    public FieldMemoryStats getFieldItemCounts() {
+        return fieldItemCounts;
     }
 
     @Override
     public void writeTo(StreamOutput out) throws IOException {
         out.writeVLong(memorySize);
         out.writeVLong(evictions);
-        out.writeOptionalWriteable(fields);
+        out.writeOptionalWriteable(fieldMemorySizes);
+        if (out.getVersion().onOrAfter(Version.V_3_4_0)) {
+            out.writeVLong(itemCount);
+            out.writeOptionalWriteable(fieldItemCounts);
+        }
     }
 
     @Override
     public XContentBuilder toXContent(XContentBuilder builder, Params params) throws IOException {
         builder.startObject(FIELDDATA);
         builder.humanReadableField(MEMORY_SIZE_IN_BYTES, MEMORY_SIZE, getMemorySize());
         builder.field(EVICTIONS, getEvictions());
+        builder.field(ITEM_COUNT, getItemCount());
+        Set<String> fields = null;
+        // We've already asserted the two key sets are the same if both are not null
+        if (fieldMemorySizes != null) {
+            fields = fieldMemorySizes.getStats().keySet();
+        } else if (fieldItemCounts != null) {
+            fields = fieldItemCounts.getStats().keySet();
+        }
         if (fields != null) {
-            fields.toXContent(builder, FIELDS, MEMORY_SIZE_IN_BYTES, MEMORY_SIZE);
+            FieldMemoryStats.toXContent(
+                builder,
+                fields,
+                FIELDS,
+                List.of(fieldMemorySizes, fieldItemCounts),
+                ORDERED_FIELD_LEVEL_API_RAW_KEYS,
+                ORDERED_FIELD_LEVEL_API_READABLE_KEYS
+            );
```

Comment:
We shouldn't be doing this. Using FieldMemoryStats logic for both itemCount and memory is what we don't want. As FieldMemoryStats is specific to memory related stats and shouldn't be used for other stuff.

Better to do it the original way? 

```
 if (fieldMemorySizes != null) {
            fieldMemorySizes.toXContent(builder, FIELDS, MEMORY_SIZE_IN_BYTES, MEMORY_SIZE);
}
if (fieldItemCounts != null) {
  fieldItemCounts.toXContent()
}
```